lib/http-server.js

```diff
@@ -148,6 +148,31 @@ function HttpServer(options) {
     });
   }
 
+  // fix named like "%" https://github.com/http-party/http-server/issues/603
+  before.push(function (req, res) {
+    function ishex(ch) {
+      return (ch >= '0' && ch <= '9') || (ch >= 'a' && ch <= 'f') || (ch >= 'A' && ch <= 'F');
+    }
+    var ind = 0 , url = req.url;
+    while ((ind = url.indexOf('%',ind)) !== -1) {
+      if ((ind + 2) >= url.length) { // 'test%' -> 'test%25' or 'test%x' -> 'test%25x'
+        url = url.substr(0,ind) + '%25' + url.substr(ind + 1);
+        break;
+      }
+      // check if %xx is hex number ,if not convert % -> %25
+      var ch1 = url.charAt(ind + 1) , ch2 = url.charAt(ind + 2);
+      if (!ishex(ch1) || !ishex(ch2)) {
+        url = url.substr(0,ind) + '%25' + url.substr(ind + 1);
```

Comment:
I'm not sure this is something the http server should be doing, shouldn't this be done on the client side?

Reply:
you are right,  i agree 。
now , i use 'encodeURI' method to test URI string , instead of the  while loop test ?  then , return 400  when URI string test failed ?